Update MIOpen version and changelog for ROCm 7.0.0 (#3899)
ROCmSoftwarePlatform/MIOpen: CHANGELOG.md

# Change Log for MIOpen

Full documentation for MIOpen is available [here](https://rocm.docs.amd.com/projects/MIOpen/en/latest/)
## MIOpen 3.5.0 for ROCm 7.0.0
### Added
* [Conv] Add misa kernels for gfx950
* [Conv] Enable split_k support for CK backward data solvers (2D)
* Add grouped convolution + activation fusion
* Add grouped convolution + bias + activation fusion
* [BatchNorm] Enable NHWC in OpenCL

> * CK can now be built inline as part of MIOpen

Reviewer: ```suggestion
* Composable Kernel (CK) can now be built inline as part of MIOpen
```

> * Swap to using median value with outliers removed for deciding on best solution to run

Reviewer: ```suggestion
* Change to using median value with outliers removed when deciding on the best solution to run
```

### Optimized
* [BatchNorm] Optimize NHWC OpenCL kernels and improve heuristics
* [RNN] Dynamic algorithm optimization
* [Conv] Eliminate redundant clearing of output buffers
* [RNN] Updated selection heuristics

### Resolved issues
* Fix segmentation fault when user specifies workspace smaller than what is required

> * Fix layout calculation logic and enable less restrictive layout selection

Reviewer: Can you add a little more detail on what the problem was (i.e. incorrect results). For example, we could say "Fixed a layout calculation logic error that returned incorrect results).

Author: I've updated this as you suggested.

I also converted the changelog to use past tense for better consistency with previous changelogs.
